pkg/networkservice/common/connect/server.go

```diff
@@ -34,22 +36,39 @@ import (
 	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
 )
 
+// Contain grpc connection and client connection associated
+type clientEntry struct {
+	clientURI       *url.URL
+	client          networkservice.NetworkServiceClient
+	cc              grpc.ClientConnInterface
+	ready           chan bool // Flag for connection is establishing
+	connections     map[string]*networkservice.Connection
+	err             error
+	closeConnection func() error
+}
+
+func (ce *clientEntry) markAsReady(state bool) {
+	ce.ready <- state
+	close(ce.ready)
+}
```

Comment:
Couldn't we just make ce.ready a chan struct{} and simply close it?

Reply:
Yes, we can :-)